c++/include/orc/Reader.hh

```diff
@@ -288,6 +288,17 @@ namespace orc {
     virtual uint64_t getCompressionSize() const = 0;
 
     /**
+     * Get ID of writer that generated the file.
+     * Current availiable Orc writers:
+     * 0 = ORC Java
+     * 1 = ORC C++
+     * 2 = Presto
+     * @param id out parameter for writer id
+     * @return true if writer id is availiable, false if otherwise
+     */
+    virtual bool getWriterId(uint32_t & id) const = 0;
```

Comment:
Ok, I see. But I am still not sure about returning ORC_FUTURE. The caller may want to obtain the exact writer ID for debugging purpose. It can choose to ignore if it wants. We just give them one more option. What do you think?